include/xsimd/arch/xsimd_neon.hpp

```diff
@@ -2296,6 +2296,112 @@ namespace xsimd
+        template <class A, uint8_t V0, uint8_t V1, uint8_t V2, uint8_t V3, uint8_t V4, uint8_t V5, uint8_t V6, uint8_t V7,
+                  uint8_t V8, uint8_t V9, uint8_t V10, uint8_t V11, uint8_t V12, uint8_t V13, uint8_t V14, uint8_t V15>
+        inline batch<int8_t, A> swizzle(batch<int8_t, A> const& self,
+                                        batch_constant<batch<uint8_t, A>, V0, V1, V2, V3, V4, V5, V6, V7,
+                                                       V8, V9, V10, V11, V12, V13, V14, V15>,
+                                        requires_arch<neon>) noexcept
+        {
+            std::array<int8_t, 16> data;
+            store_aligned(data.data(), self, A());
+            return set(batch<int8_t, A>(), A(), data[V0], data[V1], data[V2], data[V3], data[V4], data[V5], data[V6], data[V7],
+                       data[V8], data[V9], data[V10], data[V11], data[V12], data[V13], data[V14], data[V15]);
+        }
+
+        template <class A, uint16_t V0, uint16_t V1, uint16_t V2, uint16_t V3, uint16_t V4, uint16_t V5, uint16_t V6, uint16_t V7>
+        inline batch<uint16_t, A> swizzle(batch<uint16_t, A> const& self,
+                                          batch_constant<batch<uint16_t, A>, V0, V1, V2, V3, V4, V5, V6, V7>,
+                                          requires_arch<neon>) noexcept
+        {
+            std::array<uint16_t, 8> data;
+            store_aligned(data.data(), self);
+            return set(batch<uint16_t, A>(), A(), data[V0], data[V1], data[V2], data[V3], data[V4], data[V5], data[V6], data[V7]);
+        }
+
+        template <class A, uint16_t V0, uint16_t V1, uint16_t V2, uint16_t V3, uint16_t V4, uint16_t V5, uint16_t V6, uint16_t V7>
+        inline batch<int16_t, A> swizzle(batch<int16_t, A> const& self,
+                                         batch_constant<batch<uint16_t, A>, V0, V1, V2, V3, V4, V5, V6, V7>,
+                                         requires_arch<neon>) noexcept
+        {
+            std::array<int16_t, 8> data;
+            store_aligned(data.data(), self, A());
+            return set(batch<int16_t, A>(), A(), data[V0], data[V1], data[V2], data[V3], data[V4], data[V5], data[V6], data[V7]);
+        }
+
+        template <class A, uint32_t V0, uint32_t V1, uint32_t V2, uint32_t V3>
+        inline batch<uint32_t, A> swizzle(batch<uint32_t, A> const& self,
+                                          batch_constant<batch<uint32_t, A>, V0, V1, V2, V3>,
+                                          requires_arch<neon>) noexcept
+        {
+            std::array<uint32_t, 4> data;
+            store_aligned(data.data(), self, A());
+            return set(batch<uint32_t, A>(), A(), data[V0], data[V1], data[V2], data[V3]);
+        }
+
+        template <class A, uint32_t V0, uint32_t V1, uint32_t V2, uint32_t V3>
+        inline batch<int32_t, A> swizzle(batch<int32_t, A> const& self,
+                                         batch_constant<batch<uint32_t, A>, V0, V1, V2, V3>,
```

Comment:
These kernels are generic, there should be a single variadic version in generic, or am I missing something?

Reply:
Indeed